src/components/ContactsEdit.js

```diff
@@ -0,0 +1,131 @@
+import { useEffect, useState } from "react"
+import { useNavigate, useParams } from "react-router-dom"
+
+const initialFormState = {
+
+  firstName: "",
+  lastName: "",
+  street: "",
+  city: "",
+  email: "",
+  linkedin: "",
+  twitter: "",
+}
+
+
+function ContactsEdit(props) {
+  const [formState, setFormState] = useState(initialFormState)
+  const navigate = useNavigate()
+  const { setContacts, contacts } = props
+  const { id } = useParams()
+
+  useEffect(() => {
+    fetch(`http://localhost:4000/contacts/${id}`)
+      .then((response) => response.json())
+      .then((data) => {
+        console.log("data from server", data)
+        setFormState(data)
+      })
+  }, [])
+  function handleDelete() {
```

Comment:
A much cleaner approach to the delete - I was trying to do it this way for a while before I just made the inline delete function inside the .map method.